spki/src/traits.rs

```diff
@@ -120,3 +120,29 @@ impl<T: AssociatedAlgorithmIdentifier> DynAssociatedAlgorithmIdentifier for T {
         Self::ALGORITHM_IDENTIFIER.ref_to_owned()
     }
 }
+
+/// Returns `AlgorithmIdentifier` associated with the signature system.
+///
+/// Unlike AssociatedAlgorithmIdentifier this is intended to be implemented for public and/or
+/// private keys.
+pub trait SignatureAlgorithmIdentifier {
```

Comment:
This should probably be parameterized by `Params` ala #966, otherwise it's difficult to define constants

Reply:
done